### PR TITLE
Simplify Working Group and OpenGitOps project Governance

### PR DESCRIPTION
**Pull request by scottrigby**
Follow-up from #86
Resolves #40 
Co-chair election results [here](https://lists.cncf.io/g/cncf-tag-app-delivery/message/258)
Maintainers list updated https://github.com/gitops-working-group/gitops-working-group/pull/136

We're ready to update the provisional bootstrapped governance now that the
GitOps WG and sandbox project roles and relationships are more clearly defined.

- Attempted to outline known processes as explicitly as possible, while keeping
  as simple as possible given the context. Looking for feedback on this
- Bring initial GitOps WG charter into a charter.md, with some updates
- Simplify GOVERNANCE.md
- Removed stray .gitignore file

To-do:
- [x] Draft PR
- [x] Chairs consensus
- [x] Maintainers consensus 👀
- [x] Announce change on the mailing list after merging PR
- [x] Announce change in next GWG meeting
- [ ] Update "Clearly established governance" date and description in README Timeline in a new commit on this PR

Future:
- Keep charter in WG repo (with a section on how the WG is governed), and potentially move governance doc to OpenGitOps project repo, focused just on how the project is governed

Signed-off-by: Scott Rigby <scott@r6by.com>

### CODE REVIEW

File: GOVERNANCE.md
Comment (scottrigby) [inline GOVERNANCE.md:57]: This was my initial super-simple stab – assuming good will all around – though as written now it would be pretty easy to game the system if membership is self-declared and everyone can vote. We may want to do something similar to what CNF WG does, by having [interested parties](https://github.com/cncf/cnf-wg/blob/master/GOVERNANCE.md#elections) register within a certain time period before elections. I think our voting rules can be simpler though, since we aren't trying to balance 3 categorically different communities as they are. Thougths?

Comment (scottrigby) [reply GOVERNANCE.md]: I added an interested parties doc, with people and orgs listed in the initial GitOps WG charter.

**Comment by todaywasawesome** [comment]
If membership is self-declared, each company can only get two votes, and members can vote how do we deal with conflicts of self-declared membership within a company? And how do we deal with people affiliated with more than one company?

I'm not sure how this is solved in other orgs. In the CNCF, you basically have to pay dues in order to have votes on different committees etc.

**Comment by scottrigby** [comment]
Good points Dan.

> I'm not sure how this is solved in other orgs. In the CNCF, you basically have to pay dues in order to have votes on different committees etc.
> If membership is self-declared

The GitOps WG was announced as an "open group". But it also initially asked people to "register" their interest. 

I like how CNF WG governance says membership is self-declared. They then circumvent the possible problem of a company "stacking the deck" by limiting to one vote per org (so you can have 50 employees from one company involved, but that doesn't give the company any unfair advantage). To avoid unfair last minute surprises, they also specify a list of "interested parties" who can vote, which must be listed at least one week before any vote. It all sounds pretty fair to me. What do you think?

> each company can only get two votes, and members can vote ​how do we deal with conflicts of self-declared membership within a company?

My sketch had 2 votes per org (from other CNCF GOVERNANCE examples, like  [Cortex](https://github.com/cortexproject/cortex/blob/master/GOVERNANCE.md)), but now that you say this maybe that just confused things. If we follow the above process, we may want only one vote per organization as CNF governance does, to avoid any confusion.

As far as conflicts within a company goes, isn't it up to each org how they wish to cast their org vote? I'd like to offset this responsibility to the employees of that org to know how to properly communicate and decide this internally.

In general I think the idea of limiting number of votes per entity in general allows a WG to maintain independence from the companies and orgs that provide its members, and keep the direction as balanced and neutral as possible.

> And how do we deal with people affiliated with more than one company?

I'm not sure, but I'd like us to set up a simple election model that expresses a spirit of fairness and balance, and ask the group (and chairs) to use common sense around this. CNF has an entire section on [determining independence](https://github.com/cncf/cnf-wg/blob/master/GOVERNANCE.md#independence), but I didn't think we needed to get that detailed for this WG.

**Comment by scottrigby** [comment]
PS, the CNF WG [interested parties](https://github.com/cncf/cnf-wg/blob/main/interested-parties.md) doc is only a list of companies.  Whereas I kept the existing GitOps WG list of "interested parties" from the initial charter draft google doc, that also included people's names. I don't think this is necessary for governance, but it is kind of nice that people from various orgs have a way to self-declare their interest. What do you think?

**Comment by scottrigby** [comment]
Pushed changes to clarify

File: GOVERNANCE.md
Comment (murillodigital) [inline GOVERNANCE.md:36]: What do y'all think about also considering the definition of new initiatives as well as the promotion of open involvement as responsibilities for the chairs?

File: charter.md
Comment (murillodigital) [inline charter.md:16]: Would change wording to:

- Promote awareness and understanding of GitOps terminology and approaches by creating education materials, organizing events and advocating throughout the community

File: charter.md
Comment (murillodigital) [inline charter.md:32]: Need to add the OpenGitOps project website

File: GOVERNANCE.md
Comment (scottrigby) [reply GOVERNANCE.md:36]: > What do y'all think about also considering the definition of new initiatives 

I was thinking that's covered by this? WDYT
```
- Schedule discussion of proposals that have been submitted
- Partner with other WG members to establish a roadmap and manage ongoing projects
- Ask for new proposals to be made to address an identified needs
```

> as well as the promotion of open involvement as responsibilities for the chairs?

Maybe "Ensure the promotion of open involvement"? This way, it's not the sole responsibility of chairs to promote this, but only to ensure that it is in fact being promoted.

On this note, I had set up a "Communications" team in this GitHub org as an area to ask other interested people for help. For consistent messaging, I do think we need a dedicated team for this – however small – with good internal communication and clear processes. So yes, sgtm!

File: GOVERNANCE.md
Comment (todaywasawesome) [inline GOVERNANCE.md:7]: "While anyone may join and contribute, voting members must be active members of the CNCF. "

File: GOVERNANCE.md
Comment (todaywasawesome) [inline GOVERNANCE.md:57]: "Up to two individuals from organizations who are both active members of the CNCF and listed in the [interested parties](interested-parties.md) may vote."

File: GOVERNANCE.md
Comment (todaywasawesome) [inline GOVERNANCE.md:7]:
```suggestion
     While anyone may join and contribute, voting members must be active members of the CNCF.
```

File: GOVERNANCE.md
Comment (scottrigby) [inline GOVERNANCE.md:10]: @todaywasawesome I think we should specify the voting requirements in only one place so I linked below instead. Then the "membership" wording needed to be switched around to still make sense. WDTY?

```suggestion
This is an open public working group welcoming anyone who would like to join and help.
Apart from following the [Code of Conduct](CODE_OF_CONDUCT.md), there are no special requirements for membership.
Feel free to register your interest in the [interested parties](interested-parties.md) document.
Note that voting requires additional conditions (see [Elections](#elections)).
```

File: GOVERNANCE.md
Comment (scottrigby) [reply GOVERNANCE.md:37]: @murillodigital  Added your second suggestion with commit co-authorship 👍  LMK if you disagree with the first point

File: charter.md
Comment (scottrigby) [reply charter.md]: Added with commit co-authorship 👍

File: charter.md
Comment (scottrigby) [reply charter.md:33]: Added with commit co-authorship 👍

**Review by roberthstrand — APPROVED** [review]

**Review by bricef — APPROVED** [review]

**Comment by scottrigby** [comment]
Converting to draft for now. Will make changes soon per yesterday's WG meeting ⏳

**Review by lloydchang — APPROVED** [review]

**Comment by scottrigby** [comment]
Update: this pr is still going to be updated, Just had to focus on other things for a few days￼

**Comment by scottrigby** [comment]
@todaywasawesome @murillodigital I updated this PR per our last meeting. See https://github.com/scottrigby/gitops-working-group/blob/chairs/GOVERNANCE.md

The governance for OpenGitOps is not yet split out from the GitOps Working Group. I think we still have a ways to go before that. For now, the OpenGitOps GitHub org can have a GOVERNANCE file that points to this one. I want to get on the same page as chairs first, and then will move this PR out of draft mode and seek consensus by MAINTAINERS 🙂✨

**Review by murillodigital — APPROVED** [review]

**Review by rficcaglia — COMMENTED** [review]
I'm guessing this is not under CNCF?  there is a formal governance process for CNCF WG formation and governance: 

~~https://github.com/cncf/toc/blob/main/workinggroups/README.md~~

EDIT: Ah sorry I noticed just now someone else had posted a link to this.

another example from Kubernetes TOC:
https://github.com/kubernetes/community/blob/master/committee-steering/governance/wg-governance.md

I'm not saying this group needs to follow either, but...if we want to merge into one of these, then we will need to align at some point.

File: GOVERNANCE.md
Comment (murillodigital) [inline GOVERNANCE.md:83]:
```suggestion
Before each election, there will be a call for nominations on the mailing list.
```

File: GOVERNANCE.md
Comment (scottrigby) [inline GOVERNANCE.md:31]:
```suggestion
- Events Committee
  - Responsible for organizing events such as GitOpsCon, and supporting other events in the wider GitOps community (see [Timeline](#timeline))
```

File: GOVERNANCE.md
Comment (todaywasawesome) [inline GOVERNANCE.md:133]:
```suggestion
- Maintainers: Adding additional maintainers.
- Chairs: Enforcing a Code of Conduct violation by a community member.
```

**Comment by scottrigby** [comment]
👍  to changes just made by @todaywasawesome @murillodigital

**Review by todaywasawesome — APPROVED** [review]

File: GOVERNANCE.md
Comment (csand-msft) [inline GOVERNANCE.md:79]: Any thoughts about max # of consecutive terms for an individual?

**Comment by csand-msft** [comment]
Approve with one comment.

**Review by csand-msft — APPROVED** [review]
Approved.

File: GOVERNANCE.md
Comment (scottrigby) [reply GOVERNANCE.md:79]: @csand-msft LIFETIME!!! 👑 lol j/k. No strong opinion on this one 🙂

My main concern is to keep this as simple as possible, while learning what we can from other CNCF project governance docs. Ultimately whatever's best for the stability, fairness, and momentum of the group and project.

Personally my gut feeling is setting max terms isn't necessary, since requiring reelection annually should self-govern and weed out anything weird. WDYT?

**Comment by scottrigby** [comment]
@rficcaglia Hi yes, thank you!

> I'm guessing this is not under CNCF? there is a formal governance process for CNCF WG formation and governance:
> 
> ~https://github.com/cncf/toc/blob/main/workinggroups/README.md~
> 
> EDIT: Ah sorry I noticed just now someone else had posted a link to this.
> 
> another example from Kubernetes TOC:
> https://github.com/kubernetes/community/blob/master/committee-steering/governance/wg-governance.md
> 
> I'm not saying this group needs to follow either, but...if we want to merge into one of these, then we will need to align at some point.

Yes the GitOps Working Group is already a TOC-approved WG under CNCF App Delivery SIG.

The historically fun – and initially confusing – thing about this is that the "GitOps Working Group" was also the preliminary name for a CNCF Sandbox project, which was meant to be a home for lasting programs, documents and any code by WG. However, that name was just a placeholder – as discussed in the TOC Sandbox review – and since then the _sandbox project_ has been renamed to OpenGitOps. See [this discussion and answer](https://github.com/gitops-working-group/gitops-working-group/discussions/64#discussioncomment-505062) for linked details.

For historical purposes, here was the [initial GitOps WG Charter](https://docs.google.com/document/d/11EZfvB2FFI837nMmArnyv-wizsIJvc-4_xdgfoUXF4o/view#heading=h.mspta0kuuv9j).

And here is the [onboarding issue for the CNCF project by the TOC](https://github.com/cncf/toc/issues/584). Even though the issue is still titled `[PROJECT ONBOARDING] GitOps WG`, it is now really for the OpenGitOps Sandbox Project, which is led by the GitOps WG for as long as the WG is necessary.

I hope that clears some of this up.

File: GOVERNANCE.md
Comment (scottrigby) [reply GOVERNANCE.md:79]: We can also continue to iterate on governance after merging this PR.

**Comment by scottrigby** [comment]
We already have a 2/3 supermajority vote among [maintainers](https://github.com/gitops-working-group/gitops-working-group/blob/main/MAINTAINERS). But if it doesn't take _too_ long for further change requests or approval, it would be nice to have unanimity 😇  @jlbutler @chrispat WDYT?

**Review by rficcaglia — COMMENTED** [review]
my concerns were addressed!

**Comment by kmf** [comment]
If it counts for something, I approve.

**Review by chrispat — APPROVED** [review]

**Comment by jlbutler** [comment]
Sorry to join the party late. LGTM 👍

**Comment by scottrigby** [comment]
Thank you all, we have unanimity ❤️ 

We can iterate on this in follow-up PRs 🚀

**Pull request merged**
